pkg/config/mesh/mesh.go

```diff
@@ -139,3 +144,69 @@ func LoadMeshNetworksConfig(yaml string) (*meshconfig.MeshNetworks, error) {
 	// }
 	return &out, nil
 }
+
+// ReadMeshNetworks gets mesh networks configuration from a config file
+func ReadMeshNetworks(filename string) (*meshconfig.MeshNetworks, error) {
```

Comment:
I would like to keep this code in the Pilot folder so that the respective code owners can oversee any future changes. Moving them to some common pkg folder will muddy all the ownership details.  As such, the mesh pkg should not be in the shared config folder as is, since this is solely meant for use by Pilot.

Either that or edit the owners file to add the same people from Pilot into this specific folder as well.

Reply:
Not only this folder, i think applies all other folders under pkg, like pkg/adsc  pkg/bootstrap pkg/config/{...} pkg/envoy  pkg/kube, we should make sure prs including these files can be reviewed thoroughly

Reply:
Having shared code under an application directory makes it hard to reason about our dependency structure within the project. We can adjust owners appropriately in the `pkg/config/mesh` directory if you're concerned about ownership of these methods.